src/backend/columnar/columnar_metadata.c

```diff
@@ -923,31 +932,67 @@ StripeMetadataLookupRowNumber(Relation relation, uint64 rowNumber, Snapshot snap
 	ScanKeyInit(&scanKey[1], Anum_columnar_stripe_first_row_number,
 				strategyNumber, procedure, UInt64GetDatum(rowNumber));
 
-
 	Relation columnarStripes = table_open(ColumnarStripeRelationId(), AccessShareLock);
-	Relation index = index_open(ColumnarStripeFirstRowNumberIndexRelationId(),
-								AccessShareLock);
-	SysScanDesc scanDescriptor = systable_beginscan_ordered(columnarStripes, index,
-															snapshot, 2,
-															scanKey);
 
-	ScanDirection scanDirection = NoMovementScanDirection;
-	if (lookupMode == FIND_LESS_OR_EQUAL)
-	{
-		scanDirection = BackwardScanDirection;
-	}
-	else if (lookupMode == FIND_GREATER)
+	/* can't rely on indexes during pg upgrades */
+	Oid indexId = ColumnarStripeFirstRowNumberIndexRelationId();
+	bool indexOk = !IsBinaryUpgrade;
```

Comment:
Maybe not always fall-back to seq-scan, if the index exists ? This is quite possible if pg_dump decides creating the metadata index before creating columnar tables.